lib/tasks/locale.rake

```diff
@@ -43,6 +43,8 @@ namespace :locale do
   desc "Extract strings from various yaml files and store them in a ruby file for gettext:find"
   task :extract_yaml_strings => :environment do
     def update_output(string, file, output)
+      file.gsub!(pwd + '/', "")
+      file.gsub!(ManageIQ::UI::Classic::Engine.root.to_s + '/', "")
```

Comment:
Is there a way to avoid hardcoding a UI constant into here?  I understand this is a tool that a developer runs, but trying to keep components completely separate is the idea.  I'm also concerned that this handles only the classic-ui case, and doesn't include things like provider plugins.

cc @jrafanie

Reply:
Additionally, this will load the whole engine and the enormous config/routes.rb file into the process that only needs a path.  Can we solve this differently and provide a way for other plugins to register their paths for translations?

Reply:
Yes, I agree, this is not pretty. Nevertheless, the same constant has already been in the rake
task for some time since https://github.com/ManageIQ/manageiq/commit/34e1cfa8375b18efb0a864b613e046bb1bbcdce3#diff-8fcec98052e3345e335a2d0c2ecde97bR81

Right now, I need a quick solution to the problem of us including full (rather than relative)
string source paths into the gettext catalog that we create (should we need to update
the catalogs for translations soon).

In a near future, I want to change things so that yamls in UI repo will be extracted
in the UI repo (not here in core), so that these path constants won't be needed at all.